src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx

```diff
@@ -80,65 +75,80 @@ const AttritionTableRow = ({
   }, [breakdown, cohortDefinitionId, covariateSubset, sourceId]);
 
   useEffect(() => {
+    const getSizeByColumn = (hare) => {
+      const hareIndex = breakdownColumns.findIndex(
+        // eslint-disable-next-line camelcase
+        ({ concept_value }) => concept_value === hare,
+      );
+      return hareIndex > -1
+        ? breakdownColumns[hareIndex].persons_in_cohort_with_value
+        : 0;
+    };
     setAfr(getSizeByColumn('AFR'));
     setAsn(getSizeByColumn('ASN'));
     setEur(getSizeByColumn('EUR'));
     setHis(getSizeByColumn('HIS'));
   }, [breakdownColumns]);
 
-  const determineChartIcon = (rowTypeInput) => {
-    if (rowTypeInput === 'Cohort') {
-      return null;
+  const determineChartIcon = () => {
+    if (rowType === 'Covariate' || rowType === 'Outcome') {
+      if (rowObject.variable_type === 'concept') {
+        return <BarChart />;
+      }
+      return <EulerDiagram />;
```

Comment:
maybe it is best if this is an `else if (rowObject.variable_type === 'custom_dichotomous')`? And then add an `else` that will throw an error?